enterprise/admin/forms.py

```diff
@@ -69,6 +87,18 @@ def clean_email_or_username(self):
 
         return email
 
+    def clean_course(self):
+        """
+        Verify course ID and retrieve course details.
+        """
+        course_id = self.cleaned_data[self.Fields.COURSE].strip()
+        if not course_id:
+            return None
+        try:
+            return get_course_details(course_id)
+        except HttpClientError:
```

Comment:
Let's also handle HttpServerError; 500 errors raised by the API aren't handled by this block. As a result, putting in a malformed course ID like `fake-course-v1:lskdjf+kjerw+kjdsf` causes a 500 in the course API that then causes a 500 here as well.